query_optimizer/ExecutionGenerator.cpp

```diff
@@ -551,7 +554,9 @@ void ExecutionGenerator::convertHashJoin(const P::HashJoinPtr &physical_plan) {
   S::HashTable *hash_table_proto = query_context_proto_->add_join_hash_tables();
 
   hash_table_proto->set_hash_table_impl_type(
-      HashTableImplTypeProtoFromString(FLAGS_join_hashtable_type));
+      SimplifyHashTableImplTypeProto(
```

Comment:
Can we add a comment here to clearly state when the simplified hash table is selected?